packages/core/src/baseTransfer.ts

```diff
@@ -0,0 +1,87 @@
+import type { Config } from './config/config.js';
+import type { Domainlike, EvmResource, Domain, SubstrateResource } from './types.js';
+
+export interface BaseTransferParams {
+  source: Domainlike;
+  destination: Domainlike;
+  resource: string | EvmResource | SubstrateResource;
+  sourceAddress: string;
+}
+
+export abstract class BaseTransfer {
+  protected _destination: Domain;
```

Comment:
I would call it destinationDomain, transferResource, sourceDomain, etc as it is used only internally to avoid underscores.